### ISSUE TITLE
swc build graphql introspectComments and decorator error

### ISSUE THREAD

**Issue by LastKing** [nestjs/nest#11920]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

1. use `nest-cli.json`  `compilerOptions.builder.swc:true`
2. modify Author code to 
    ```
    @ObjectType()
    export class Author {
      /**
       * Test
       */
      @Field((type) => Int)
      id: number;
   }
    ```
4. generator schema.gql
    ``` 
    type Author {
      id: Int!
      firstName: String
      lastName: String
     posts: [Post!]!
    }
    ```

### Minimum reproduction code

https://github.com/toonew-issue/nestjs-swc-graphql-comments

### Steps to reproduce

1. webstorm debug run  `start:debug`

### Expected behavior

expected generate
```
type Author {
  """Test"""
  id: Int!
  firstName: String
  lastName: String
  posts: [Post!]!
}
```

### Package

- [ ] I don't know. Or some 3rd-party package
- [X] <code>@nestjs/common</code>
- [X] <code>@nestjs/core</code>
- [ ] <code>@nestjs/microservices</code>
- [ ] <code>@nestjs/platform-express</code>
- [ ] <code>@nestjs/platform-fastify</code>
- [ ] <code>@nestjs/platform-socket.io</code>
- [ ] <code>@nestjs/platform-ws</code>
- [ ] <code>@nestjs/testing</code>
- [ ] <code>@nestjs/websockets</code>
- [ ] Other (see below)

### NestJS version

10.0.3

### Packages versions

```json
{
  "name": "nestjs-study",
  "version": "0.0.1",
  "description": "",
  "author": "",
  "private": true,
  "license": "UNLICENSED",
  "scripts": {
    "build": "nest build",
    "format": "prettier --write \"src/**/*.ts\" \"test/**/*.ts\"",
    "start": "nest start",
    "start:dev": "nest start --watch",
    "start:debug": "nest start --debug --watch",
    "start:prod": "node dist/main",
    "lint": "eslint \"{src,apps,libs,test}/**/*.ts\" --fix",
    "test": "jest",
    "test:watch": "jest --watch",
    "test:cov": "jest --coverage",
    "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
    "test:e2e": "jest --config ./test/jest-e2e.json"
  },
  "dependencies": {
    "@apollo/server": "^4.7.5",
    "@nestjs/apollo": "^12.0.4",
    "@nestjs/common": "^10.0.3",
    "@nestjs/core": "^10.0.3",
    "@nestjs/graphql": "^12.0.5",
    "@nestjs/platform-express": "^10.0.3",
    "graphql": "^16.7.1",
    "reflect-metadata": "^0.1.13",
    "rxjs": "^7.8.1"
  },
  "devDependencies": {
    "@nestjs/cli": "^10.0.5",
    "@nestjs/schematics": "^10.0.1",
    "@nestjs/testing": "^10.0.3",
    "@swc/cli": "^0.1.62",
    "@swc/core": "^1.3.66",
    "@types/express": "^4.17.17",
    "@types/jest": "^29.5.2",
    "@types/node": "^20.3.2",
    "@types/supertest": "^2.0.12",
    "@typescript-eslint/eslint-plugin": "^5.60.1",
    "@typescript-eslint/parser": "^5.60.1",
    "eslint": "^8.43.0",
    "eslint-config-prettier": "^8.8.0",
    "eslint-plugin-prettier": "^4.2.1",
    "jest": "^29.5.0",
    "prettier": "^2.8.8",
    "source-map-support": "^0.5.21",
    "supertest": "^6.3.3",
    "ts-jest": "^29.1.0",
    "ts-loader": "^9.4.3",
    "ts-node": "^10.9.1",
    "tsconfig-paths": "^4.2.0",
    "typescript": "^5.1.5"
  },
  "jest": {
    "moduleFileExtensions": [
      "js",
      "json",
      "ts"
    ],
    "rootDir": "src",
    "testRegex": ".*\\.spec\\.ts$",
    "transform": {
      "^.+\\.(t|j)s$": "ts-jest"
    },
    "collectCoverageFrom": [
      "**/*.(t|j)s"
    ],
    "coverageDirectory": "../coverage",
    "testEnvironment": "node"
  }
}
```

### Node.js version

18.16.0

### In which operating systems have you tested?

- [ ] macOS
- [X] Windows
- [ ] Linux

**Comment by kamilmysliwiec** [comment]
Fixed in @nestjs/graphql@12.0.6

**Comment by LastKing** [comment]
I update the dependencies.
1. when i run  `nest build`, `meatadata.ts` update ,but `shema.gql` don't change.
2. when i run `nest start` throw error
   ![image](https://github.com/nestjs/nest/assets/8540374/2dae9504-3c54-4682-8371-01525fbd76e3)

I update mini reproduction code repo.

Thank you!~!~

**Comment by kamilmysliwiec** [comment]
Fixed, thanks for reporting! (v12.0.7)
